lsp-completion.el

```diff
@@ -501,7 +502,17 @@ The MARKERS and PREFIX value will be attached to each candidate."
           ;; always be shown
           ((equal action 'lambda) nil)
           ;; retrieve candidates
-          ((equal action t) (funcall candidates))))
+          ((equal action t)
+           (if (string-empty-p probe)
+               ;; when triggered via `lsp-passthrough'
+               (funcall candidates)
```

Comment:
Checking for an empty probe is an unnecessary optimization here. It may lead to problems later on since the returned list is mutated (assuming the list is cached).

Reply:
We don't cache that list, rather we cache the list that's the evaluation of `(funcall candidates)`.
But yes, this optimization might not need.

Reply:
If you cache the list returned by funcall candidates the optimization is even incorrect, since the frontend may mutate the returned list. But maybe I misunderstood you.